lab_1/main.py

```diff
@@ -87,16 +177,75 @@ def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> li
     right_n = 3
     --> [['man', 'is'], ['dog, 'cat']]
     """
-    pass
+    all_adjacent_words = []
+    context = []
+    test1 = isinstance(tokens, list)
+    test2 = isinstance(word, str)
+    test3 = isinstance(left_n, int)
+    test4 = isinstance(right_n, int)
+    if not (test1 and test2 and word in tokens and test3 and test4):
+        return []
+    if isinstance(left_n, bool) or isinstance(right_n, bool):
+        return []
+    if right_n > 0 and left_n > 0:
+        concordance = get_concordance(tokens, word, left_n, right_n)
+        num = len(concordance)
+        text = concordance[0]
+        context.append(text[0])
+        context.append(text[-1])
+        all_adjacent_words.append(context)
+        if num > 1:
+            concordance.remove(text)
```

Comment:
you do not really need all of this, you can iterate over a concordance itself using for